syntax = "proto3";

package ExternalKownledge;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "UUID.proto";

service ExternalKnowledgeController {
    rpc List(ExternalKnowledgeListRequest) returns (stream ExternalKnowledge) {}
    rpc Create(ExternalKnowledge) returns (ExternalKnowledge) {}
    rpc Retrieve(ExternalKnowledgeRetrieveRequest) returns (ExternalKnowledge) {}
    rpc Update(ExternalKnowledge) returns (ExternalKnowledge) {}
    rpc Destroy(ExternalKnowledge) returns (google.protobuf.Empty) {}
}

message ExternalKnowledge {
    UUID id = 1;
    string subject = 2;
    string chapter = 3;
    string content = 4;
    repeated float content_embedding = 5;
    repeated float subject_embedding = 6;
    repeated float chapter_embedding = 7;
    google.protobuf.Timestamp created_at = 8;
    google.protobuf.Timestamp updated_at = 9;
}

message ExternalKnowledgeListRequest {
}

message ExternalKnowledgeRetrieveRequest {
    string id = 1;
}
